core/src/main/java/org/apache/calcite/util/format/FormatElementEnum.java

```diff
@@ -81,6 +81,13 @@ public enum FormatElementEnum implements FormatElement {
       sb.append(work.eeeFormat.format(date));
     }
   },
+  E("d", "The day of the month as a decimal number (01-31) left-padded with space") {
```

Comment:
Hi @mihaibudiu, all changes looks good to me.

We can replace `(01-31)` in the comment to `(1-31)`, and maybe mention "single digits left-padded with space" to avoid confusion and align with BQ docs.